docs/L_06_Exceptions.md

```diff
@@ -0,0 +1,75 @@
+### Wyjątki
+Wyjątki: coś się spieprzyło, nie działa, nie wiem co z tym zrobić, niech się ktoś tym zajmie.
+
+**Takie bardzo silne go to z przekazaniem parametru i zamknięciem wszystkich klamer. Wielokrotny return**
+
+Wyjątki są w każdym współczesnym języku
```

Comment:
Padło pytanie o podobieństwo do C++: "Throw by Value, Catch by Reference"

Tego mechanizmu nie ma, bo nie można rzucić "by value", bo i tak wszystko jest wskaźnikiem do obiektu.  
Ale też: w Javie jest inny mechanizm rzutowania. Tam łapaliśmy przez referencję, żeby C++ nie spróbowało niejawnie przerzutować złapanego wyjątku na inny typ. W Javie jest tak, że jak jakiś obiekt zrzutujesz na coś (choćby na Object) to on i tak tam w środku, w obiekcie nadal wie "kim naprawdę jest".  
O taki kod:
```Java
        Person person = new Person("Gru", "Dru", 123L);
        Object o = (Object)person;
        System.out.println(o.getClass());
```
pokaże, że o jest nadal `Person` a nie `Object`. To ryzyko z C++ (niejawnej zmiany typu w locie) nie zachodzi